Telegraf 0.13.1 panic in Disk Plugin

## Bug report

### System info:

Telegraf 0.13.1
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=12.04
DISTRIB_CODENAME=precise
DISTRIB_DESCRIPTION="Ubuntu 12.04.5 LTS"

Relevant Config:
```
#
# INPUTS:
#
[[inputs.cpu]]
  percpu = true
  totalcpu = true
[[inputs.disk]]
[[inputs.diskio]]
[[inputs.kernel]]
[[inputs.mem]]
[[inputs.net]]
[[inputs.netstat]]
[[inputs.processes]]
[[inputs.puppetagent]]
[[inputs.swap]]
[[inputs.system]]
```

Error Message:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x63fd09]

goroutine 98 [running]:
panic(0x107ebc0, 0xc820010080)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/system.(*systemPS).DiskUsage(0x1c78550, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/ps.go:87 +0x829
github.com/influxdata/telegraf/plugins/inputs/system.(*DiskStats).Gather(0xc82005c1e0, 0x7f369a02ba88, 0xc8203f8000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/disk.go:44 +0xd9
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc8203f40c0, 0xc82005f110, 0xc8203f8000)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:174 +0x73
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:175 +0xe0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -44,6 +44,7 @@ time before a new metric is included by the plugin.
 - [#1258](https://github.com/influxdata/telegraf/pull/1258): Fix potential kernel plugin integer parse error.
 - [#1268](https://github.com/influxdata/telegraf/pull/1268): Fix potential influxdb input type assertion panic.
 - [#1283](https://github.com/influxdata/telegraf/pull/1283): Still send processes metrics if a process exited during metric collection.
+- [#1297](https://github.com/influxdata/telegraf/issues/1297): disk plugin panic when usage grab fails.
 
 ## v0.13.1 [2016-05-24]
 

--- a/plugins/inputs/system/ps.go
+++ b/plugins/inputs/system/ps.go
@@ -84,10 +84,10 @@ func (s *systemPS) DiskUsage(
 		mountpoint := os.Getenv("HOST_MOUNT_PREFIX") + p.Mountpoint
 		if _, err := os.Stat(mountpoint); err == nil {
 			du, err := disk.Usage(mountpoint)
-			du.Path = p.Mountpoint
 			if err != nil {
 				return nil, err
 			}
+			du.Path = p.Mountpoint
 			// If the mount point is a member of the exclude set,
 			// don't gather info on it.
 			_, ok := fstypeExcludeSet[p.Fstype]

